fuse/src/main/java/tachyon/fuse/TachyonFuseFs.java

```diff
@@ -0,0 +1,666 @@
+    final int openFlag = flags & 3;
+    if (openFlag != O_WRONLY.intValue()) {
+      OpenFlags flag = OpenFlags.valueOf(openFlag);
+      LOG.error("Passed a {} flag to create(). Files can only be created in O_WRONLY mode ({})",
+          flag.toString(), path);
+      return -ErrorCodes.EACCES();
+    }
+
+    try {
+
+      final OpenFileEntry ofe =
+          new OpenFileEntry(null, mTFS.getOutStream(turi));
+      LOG.debug("Tachyon OutStream created for {}", path);
+
+      synchronized (mOpenFiles) {
+        if (mOpenFiles.size() == MAX_OPEN_FILES) {
+          LOG.error("Cannot open {}: too many open files", turi);
+          return -ErrorCodes.EMFILE();
+        }
+        mOpenFiles.put(mNextOpenFileId, ofe);
+        fi.fh.set(mNextOpenFileId);
+
+        // Assuming I will never wrap around (2^64 open files are quite a lot anyway)
+        mNextOpenFileId += 1;
+      }
+      LOG.debug("{} created and opened in O_WRONLY mode", path);
+
+    } catch (FileAlreadyExistsException e) {
+      LOG.debug("File {} already exists", turi, e);
+      return -ErrorCodes.EEXIST();
+    } catch (IOException e) {
+      LOG.error("IOException on {}", path, e);
+      return -ErrorCodes.EFAULT();
+    } catch (TachyonException e) {
+      LOG.error("TachyonException on {}", path, e);
+      return -ErrorCodes.EFAULT();
+    } catch (Throwable e) {
+      LOG.error("Unexpected exception on {}", path, e);
+      return -ErrorCodes.EFAULT();
+    }
+
+    return 0;
+  }
+
+  /**
+   * Flushes cached data on Tachyon.
+   *
+   * Called on explicit sync() operation or at close().
+   * @param path The path on the FS of the file to close
+   * @param fi FileInfo data struct kept by FUSE
+   * @return 0 on success, a negative value on error
+   */
+  @Override
+  public int flush(String path, FuseFileInfo fi) {
+    LOG.trace("flush({})", path);
+    final long fd = fi.fh.get();
+    OpenFileEntry oe = null;
+    synchronized (mOpenFiles) {
+      oe = mOpenFiles.get(fd);
+    }
+    if (oe == null) {
+      LOG.error("Cannot find fd for {} in table", path);
+      return -ErrorCodes.EBADFD();
+    }
+    if (oe.getOut() != null) {
+      try {
+        oe.getOut().flush();
+      } catch (IOException e) {
+        return -ErrorCodes.EIO();
+      }
+    } else {
+      LOG.debug("Not flushing: {} was not open for writing", path);
+    }
+    return 0;
+  }
+
+  /**
+   * Retrieves file attributes.
+   * @param path The path on the FS of the file
+   * @param stat FUSE data structure to fill with file attrs
+   * @return 0 on success, negative value on error
+   */
+  @Override
+  public int getattr(String path, FileStat stat) {
+    final TachyonURI turi = mPathResolverCache.getUnchecked(path);
+    LOG.trace("getattr({}) [Tachyon: {}]", path, turi);
+    try {
+      final TachyonFile tf = mTFS.openIfExists(turi);
+      if (tf == null) {
+        return -ErrorCodes.ENOENT();
+      } else {
```

Comment:
NIT: since we have returned in the if-case, we probably don't need this else clause, and I feel this can ease reading a little bit (since you don't need to look for the end of else-clause).